### ISSUE TITLE
[Rogue - Vanish] Vanish doesn't get me out of combat when used alone

### ISSUE THREAD

**Issue by Elexias** [chromiecraft/chromiecraft#2919]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

I tried to survive in MC raid, by using Vanish, when I was only one alive, and bosses kept attacking me after.
![WoWScrnShot_020122_210723](https://user-images.githubusercontent.com/80590226/152050978-0ab26e8d-638e-4c5c-a93f-507f8039e045.jpg)

### Expected Blizzlike Behaviour

Should reset properly like it used to work before

### Steps to reproduce the problem

1. Let everyone from raid die
2. Use vanish
3. Bosses keep reattacking you after you vanish

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/6f20b9ece71984f1a11469e65ede8accda05653e

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Elexias** [comment]
You can see that my vanish buff is on me means I used it, you can also check vanish CD

**Comment by IntelligentQuantum** [comment]
Tested on local ac works fine.

**Comment by Elexias** [comment]
@IntelligentQuantum Please give us more information about your testing, were you in raid group? How many people you had with yourself, what threat on list you were, what talents you used? Testing on AC is fine if you can copy real raid situation. Else I don't accept your testing.

**Comment by IntelligentQuantum** [comment]
Talent combat
6 pp

That's why I put the Needs second opinion label, maybe I'm wrong :wink:

**Comment by Salehyn** [comment]
Cannot reproduce. Vanish will properly reset bosses as long as you're the only one left in combat. 
Tested on the CC PTR in a raid group with an alt who would get killed by the boss before me. Talents or the lack thereof, and further corpses in the raid have no reason to affect the outcome. "Threat list" doesn't have any variations when you're the only person left alive.

Note: Vanish has a small chance to be resisted, just like feign death. Vanish can also fail if used in close melee range due to lag, basically the enemy might land a melee hit on you after you vanished, which immediately breaks stealth. This has a higher chance of happening on boss fights with adds since they melee you at different intervals.
Vanish also won't work if there is anyone else still alive in any shape or form to keep boss in combat. I do see a hunter and a priest both alive on your screenshot, so you clearly weren't alone. Even if the hunter was feigned when you used your vanish, the next "combat pulse" would put you back on the boss' threat list, which means he will go for you, vanished or not. 

TL;DR this seems to be working as intended and you experienced a once-off vanish fail, either due to resist or player error.
